src/Hal.h

```diff
@@ -0,0 +1,42 @@
+#include <stdint.h>
+#include <stddef.h>
+#if !defined(_RADIOLIB_HAL_H)
+#define _RADIOLIB_HAL_H
+
+class Hal {
+  public:
+    const uint8_t GpioModeInput;
+    const uint8_t GpioModeOutput;
+    const uint8_t GpioLevelLow;
+    const uint8_t GpioLevelHigh;
+    const uint8_t GpioInterruptRising;
+    const uint8_t GpioInterruptFalling;
```

Comment:
I don't think it's safe to use uint8_t as the generic "pin type", I would propose to change this to uint32_t as most of the platforms RadioLib runs on are 32-bit.

Reply:
I don't understand , why wouldn't it be safe?

Reply:
> I don't think there is a platform with more than 255 pins

But the value can contain more information than just a pin number. In fact most MCUs address GPIO pins using port and pin type. IIRC some STM32 Arduino cores use the number to save both GPIO port as well as GPIO pin, which may add up to more than 8 bits.